policy-test/tests/accepted_http_route_status.rs

```diff
@@ -0,0 +1,216 @@
+            namespace: server.namespace(),
+            name: server.name_unchecked(),
+            section_name: None,
+            port: None,
+        }];
+
+        // Create a route that references the Server resource.
+        let _route = create(
+            &client,
+            mk_inbound_route(&ns, "test-accepted-route", Some(srv_ref)),
+        )
+        .await;
+        // Wait until route is updated with a status
+        let route_status = {
+            let status = await_route_status(&client, &ns, "test-accepted-route").await;
+            status
+                .parents
+                .get(0)
+                .expect("must have at least one parent status")
+                .clone()
+        };
+
+        // Check status references to parent we have created
+        let parent_ref = &route_status.parent_ref;
+        assert_eq!(parent_ref.group, Some("policy.linkerd.io".to_string()));
+        assert_eq!(parent_ref.kind, Some("Server".to_string()));
+        assert_eq!(parent_ref.name, server.name_unchecked());
+
+        // Check status is accepted with a status of 'True'
+        let cond = route_status
+            .conditions
+            .iter()
+            .find(|cond| cond.type_ == "Accepted".to_string())
+            .expect("must have at least one 'Accepted' condition");
+        assert_eq!(cond.status, "True".to_string());
+        assert_eq!(cond.reason, "Accepted".to_string())
+    })
+    .await;
+}
+
+#[tokio::test(flavor = "current_thread")]
+async fn inbound_accepted_no_parent() {
+    with_temp_ns(|client, ns| async move {
+        // Create a reference to a Server that doesn't exist.
+        let srv_ref = vec![k8s::policy::httproute::ParentReference {
+            group: Some("policy.linkerd.io".to_string()),
+            kind: Some("Server".to_string()),
+            namespace: Some(ns.clone()),
+            name: "test-inexistent-server".to_string(),
+            section_name: None,
+            port: None,
+        }];
+
+        // Create the route
+        let _route = create(
+            &client,
+            mk_inbound_route(&ns, "test-no-parent-route", Some(srv_ref)),
+        )
+        .await;
+        let route_status = {
+            let status = await_route_status(&client, &ns, "test-no-parent-route").await;
+            status
+                .parents
+                .get(0)
+                .expect("must have at least one parent status")
+                .clone()
+        };
+        let cond = route_status
+            .conditions
+            .iter()
+            .find(|cond| cond.type_ == "Accepted".to_string())
+            .expect("must have at least one 'Accepted' condition");
+
+        // Route shouldn't be accepted
+        assert_eq!(cond.status, "False".to_string());
+        assert_eq!(cond.reason, "NoMatchingParent".to_string());
+    })
+    .await;
+}
+
+#[tokio::test(flavor = "current_thread")]
+async fn inbound_accepted_reconcile() {
```

Comment:
is it worth having the opposite test as well where we delete a Server and assert that the accepted status changes from true to false?

Reply:
I thought it would be a bit redundant but in hindsight I think it's good to test our index processes delete events accordingly.